velox/expression/SimpleFunctionAdapter.h

```diff
@@ -270,6 +288,9 @@ class SimpleFunctionAdapter : public VectorFunction {
       ApplyContext& applyContext,
       bool allNotNull,
       const TReader&... readers) const {
+    bool callNullFree = FUNC::is_default_contains_nulls_behavior ||
```

Comment:
shall we move up the logic where check contains nullls here instead of multiple places down:
and make callNullFree means callNullFree.

This way the whole control flow is clear in one location 
```
bool CallNullFree = false;
if(FUNC::udf_has_callNullFree){
  if(FUNC::is_default_contains_nulls_behavior){
    bool containNull = applyContext.mayHaveNullsRecursive() &&  readers.contansNulls(...);
     if(containNull){
        // return null no function need to be called.
     }else{
     callNullFree = true; 
     }
  }else{
       // not default nulls behaviour 
        if(applyContext.mayHaveNullsRecursive){
            callNullFree = false; // we invoke call or callnullbale
      }else{
         callNullFree = true; 
      }
   }
  }
}
```

Reply:
Discussed offline, the challenge here is that readers.contansNulls needs to be applied to each row independently, while the other checks in here are per batch.